modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/redshift/ShredModel.scala

```diff
@@ -145,29 +145,49 @@ object ShredModel {
                 case ColumnType.RedshiftInteger => NoChanges.asRight
                 case ColumnType.RedshiftBigInt => NoChanges.asRight
                 case ColumnType.RedshiftDouble => NoChanges.asRight
-                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) > 5 => NoChanges.asRight
+                case ColumnType.RedshiftVarchar(oldSize) if oldSize >= 5 => NoChanges.asRight
+                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) >= 5 => NoChanges.asRight
                 case _ => IncompatibleTypes(oldCol, newCol).asLeft.toEitherNel
               }
               case ColumnType.RedshiftInteger => oldType match {
                 case ColumnType.RedshiftInteger => NoChanges.asRight
                 case ColumnType.RedshiftBigInt => NoChanges.asRight
                 case ColumnType.RedshiftDouble => NoChanges.asRight
-                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) > 10 => NoChanges.asRight
+                case ColumnType.RedshiftVarchar(oldSize) if oldSize >= 10 => NoChanges.asRight
+                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) >= 10 => NoChanges.asRight
                 case _ => IncompatibleTypes(oldCol, newCol).asLeft.toEitherNel
               }
               case ColumnType.RedshiftBigInt => oldType match {
                 case ColumnType.RedshiftBigInt => NoChanges.asRight
                 case ColumnType.RedshiftDouble => NoChanges.asRight
-                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) > 19 => NoChanges.asRight
+                case ColumnType.RedshiftVarchar(oldSize) if oldSize >= 19 => NoChanges.asRight
+                case ColumnType.RedshiftDecimal(precision, scale) if precision.getOrElse(0) - scale.getOrElse(0) >= 19 => NoChanges.asRight
                 case _ => IncompatibleTypes(oldCol, newCol).asLeft.toEitherNel
               }
               case ColumnType.RedshiftDecimal(new_precision, new_scale) => oldType match {
                 case ColumnType.RedshiftDouble => NoChanges.asRight
+                case ColumnType.RedshiftVarchar(oldSize) if oldSize >= new_precision.getOrElse(0) => NoChanges.asRight
```

Comment:
I'm not sure about the `getOrElse(0)`.  Is it possible that the old type is like `VARCHAR(2)` and the new type is RedshiftDecimal with no precision.... which I think means it allows numbers like `12345`.  Wouldn't this mean the new value cannot be loaded to the old column type?

Reply:
decimal with no precision has default of 18

https://docs.aws.amazon.com/redshift/latest/dg/r_Numeric_types201.html